internal/services/azure_offering_template_service.go

```diff
@@ -0,0 +1,49 @@
+package services
+
+import (
+	"net/http"
+	"text/template"
+
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+)
+
+var LighthouseTemplateFilePath = "./docs/lighthouse_template.json"
+
+type azureOfferingParams struct {
+	// OfferingDefaultName that Customer can change while deploying the offering
+	OfferingDefaultName string
+
+	// OfferingDefaultDescription describing the offering, can be changed by Customer while deploying
+	OfferingDefaultDescription string
+
+	// TenantID of the offering tenant (Azure account)
+	TenantID string
+
+	// EnterpriseAppID of the App that will act as an offering Principal
+	EnterpriseAppID string
+
+	// EnterpriseAppName of the offering principal - the display name
+	EnterpriseAppName string
+}
+
+func AzureOfferingTemplate(w http.ResponseWriter, r *http.Request) {
+	params := azureOfferingParams{
+		OfferingDefaultName:        "Red Hat Hybrid cloud console",
+		OfferingDefaultDescription: "Allows Red Hat to upload images and deploy Virtual Machines from Hybrid cloud console",
+		TenantID:                   config.Azure.TenantID,
+		EnterpriseAppID:            config.Azure.ClientID,
+		EnterpriseAppName:          config.Azure.ClientName,
+	}
+
+	template, err := template.New("lighthouse_template.json").ParseFiles(LighthouseTemplateFilePath)
+	if err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "failed to fetch Azure template", err))
+		return
+	}
+
+	if err = template.Execute(w, params); err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "failed to render the Azure template", err))
+		return
+	}
+}
```

Comment:
Can you also log an INFO message that we served this? Just curious, there will not be much traffic but it is a public service so I am interested in tracking this.

Reply:
Isn't this logged automatically by the logger middleware? :thinking:

Reply:
Yes if you keep this in the API mount function, I was sort of assuming this needs to go to the RootApi mount function where there are no middlewares (I think).